python/ray/tune/syncer.py

```diff
@@ -758,6 +837,8 @@ def on_checkpoint(
             )
 
     def wait_for_all(self):
+        self._cleanup_trial_sync_processes()
+
         failed_syncs = {}
```

Comment:
say the last trial has a huge blob of data to send to driver and that takes a while. IIUC, we run though this `_cleanup_trial_sync_processes` with best effort, but then the last trial is counted as failed syncing?

Reply:
The cleanup only cleans up sync processes that were scheduled to be removed. In your case, the `wait_for_all` logic still holds - we wait until the process finishes, and it may or may not fail.

The `_cleanup_trial_sync_processes` is not strictly necessary here, but I would still keep it to remove stale syncers before trying to await non-stale ones.

Reply:
ah right, I missed `sync_process.wait()` on line 845. So we will do wait on the last trial there.